docs/tutorials/multimodal/clip_embedding.md

```diff
@@ -0,0 +1,115 @@
+# CLIP in AutoMM - Extract Embeddings 
+:label:`sec_automm_clip_embedding`
+
+We have shown CLIP's amazing capability in performing zero-shot image classification in our previous tutorial :ref:`sec_automm_clip_zeroshot_imgcls`. Thanks to the contrastive loss objective and trained on millions of image-text pairs, CLIP learns good embeddings for both vision and language, and their connections. Hence, another important use case of CLIP is to extract embeddings for retrieval, matching, ranking kind of tasks.
+
+In this tutorial, we will show you how to use AutoGluon to extract embeddings from CLIP, and then use it for a retrieval problem. 
+
+
+## Extract Embeddings
+
+CLIP has a two-tower architecture, which means it has two encoders: one for image, the other for text. An overview of CLIP model can be seen in the diagram below. Left shows its pre-training stage, and Right shows its zero-shot predicton stage. By computing the cosine similarity scores between one image embedding and all the text images, we pick the text which has the highest similarity as the prediction.
+
+![CLIP](https://github.com/openai/CLIP/raw/main/CLIP.png)
+
+Given the two encoders, we can extract image embeddings, or text embeddings. And most importantly, embedding extraction can be done offline, only similarity computation needs to be done online. So this means good scalability. 
+
+First, let's download some images. These images are from [COCO datasets](https://cocodataset.org/#home).
+
+```{.python .input}
+from autogluon.multimodal import download
+
+urls = ['http://farm4.staticflickr.com/3179/2872917634_f41e6987a8_z.jpg',
+        'http://farm4.staticflickr.com/3629/3608371042_75f9618851_z.jpg',
+        'https://farm4.staticflickr.com/3795/9591251800_9c9727e178_z.jpg',
+        'http://farm8.staticflickr.com/7188/6848765123_252bfca33d_z.jpg',
+        'https://farm6.staticflickr.com/5251/5548123650_1a69ce1e34_z.jpg']
+
+image_paths = [download(url) for url in urls]
+print(image_paths)
+```
+
+Let's extract some image embedding from the CLIP vision encoder,
+
+```{.python .input}
+from autogluon.multimodal import MultiModalPredictor
+predictor = MultiModalPredictor(hyperparameters={"model.names": ["clip"]}, problem_type="zero_shot")
+
+# extract image embeddings.
+image_embeddings = predictor.extract_embedding({"image": image_paths})
+for k, v in image_embeddings.items():
```

Comment:
Since we know there is only one key `image` here, we may just use `image_embeddings["image"]`.